### PR TITLE
Add inline example for grdvolume

### PR DESCRIPTION
**Pull request by willschlitzer**
This PR adds an inline code example to the docstring for `grdvolume`.

Addresses #1686

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/grdvolume.py [hunk 1]
```diff
@@ -75,6 +75,21 @@ def grdvolume(grid, output_type="pandas", outfile=None, **kwargs):
         - :class:`pandas.DataFrame` or :class:`numpy.ndarray` if ``outfile``
           is not set (depends on ``output_type`` [Default is
           class:`pandas.DataFrame`])
+
+    Example
+    -------
+    >>> import pygmt  # doctest: +SKIP
+    >>> # Load a grid of @earth_relief_30m data, with an x-range of 10 to 30,
+    >>> # and a y-range of 15 to 25
```
Comment (michaelgrund) [inline pygmt/src/grdvolume.py:83]:
```suggestion
    >>> # Load a grid of @earth_relief_30m data, with an x-range of 10 to 30 degrees,
    >>> # and a y-range of 15 to 25 degrees
```

File: pygmt/src/grdvolume.py [hunk 2]
```diff
@@ -75,6 +75,21 @@ def grdvolume(grid, output_type="pandas", outfile=None, **kwargs):
...
+    -------
+    >>> import pygmt  # doctest: +SKIP
+    >>> # Load a grid of @earth_relief_30m data, with an x-range of 10 to 30 degrees,
+    >>> # and a y-range of 15 to 25 degrees
+    >>> grid = pygmt.datasets.load_earth_relief(
+    ...     resolution="30m", region=[10, 30, 15, 25]
+    ... )  # doctest: +SKIP
+    >>> # Create a numpy array for the volume of slices in the grid file; set
+    >>> # the minimum z-value to 200, the maximum to 400,
+    >>> # and the interval to 10
```
Comment (seisman) [inline pygmt/src/grdvolume.py:89]:
```suggestion
    >>> # and the interval to 50
```

**Review by seisman — APPROVED** [review]
Looks good to me except a typo.

**Comment by seisman** [comment]
Need to fix the styling issue before merge.

File: pygmt/src/grdvolume.py [hunk 3]
```diff
@@ -75,6 +75,21 @@ def grdvolume(grid, output_type="pandas", outfile=None, **kwargs):
         - :class:`pandas.DataFrame` or :class:`numpy.ndarray` if ``outfile``
           is not set (depends on ``output_type`` [Default is
           class:`pandas.DataFrame`])
+
+    Example
+    -------
+    >>> import pygmt  # doctest: +SKIP
```
Comment (seisman) [inline pygmt/src/grdvolume.py:81]:
```suggestion
    >>> import pygmt
```

File: pygmt/src/grdvolume.py [hunk 4]
```diff
@@ -75,6 +75,21 @@ def grdvolume(grid, output_type="pandas", outfile=None, **kwargs):
...
+    >>> # Load a grid of @earth_relief_30m data, with an x-range of 10 to 30 degrees,
+    >>> # and a y-range of 15 to 25 degrees
+    >>> grid = pygmt.datasets.load_earth_relief(
+    ...     resolution="30m", region=[10, 30, 15, 25]
+    ... )  # doctest: +SKIP
```
Comment (seisman) [inline pygmt/src/grdvolume.py:86]:
```suggestion
    >>> grid = pygmt.datasets.load_earth_relief(
    ...     resolution="30m", region=[10, 30, 15, 25]
    ... )
```

File: pygmt/src/grdvolume.py [hunk 5]
```diff
@@ -75,6 +75,21 @@ def grdvolume(grid, output_type="pandas", outfile=None, **kwargs):
...
+    >>> # Create a numpy array for the volume of slices in the grid file; set
+    >>> # the minimum z-value to 200, the maximum to 400,
+    >>> # and the interval to 10
+    >>> output_array = pygmt.grdvolume(
+    ...     grid=grid, contour=[200, 400, 50], output_type="numpy"
+    ... )  # doctest: +SKIP
```
Comment (seisman) [inline pygmt/src/grdvolume.py:92]:
```suggestion
    >>> output_array = pygmt.grdvolume(
    ...     grid=grid, contour=[200, 400, 50], output_type="numpy"
    ... )
```

**Review by seisman — APPROVED** [review]

File: pygmt/src/grdvolume.py [hunk 6]
```diff
@@ -75,6 +77,22 @@ def grdvolume(grid, output_type="pandas", outfile=None, **kwargs):
         - :class:`pandas.DataFrame` or :class:`numpy.ndarray` if ``outfile``
           is not set (depends on ``output_type`` [Default is
           class:`pandas.DataFrame`])
```
Comment (meghanrjones) [inline pygmt/src/grdvolume.py:79]:
```suggestion
          :class:`pandas.DataFrame`])
```

File: pygmt/src/grdvolume.py [hunk 7]
```diff
@@ -74,7 +76,23 @@ def grdvolume(grid, output_type="pandas", outfile=None, **kwargs):
...
+    >>> # degrees, and a y-range of 15 to 25 degrees
+    >>> grid = pygmt.datasets.load_earth_relief(
+    ...     resolution="30m", region=[10, 30, 15, 25]
+    ... )
+    >>> # Create a numpy ndarray that contains the contour, area, volume, and
+    >>> # maximum mean height above the plane specified by the given contour
+    >>> # and below the surface; set the minimum contour z-value to 200, the
+    >>> # maximum to 400, and the interval to 50.
+    >>> output_array = pygmt.grdvolume(
+    ...     grid=grid, contour=[200, 400, 50], output_type="numpy"
```
Comment (weiji14) [inline pygmt/src/grdvolume.py:94]: Why not use a `pandas` output type?

```suggestion
    >>> # Create a pandas dataframe that contains the contour, area, volume,
    >>> # and maximum mean height above the plane specified by the given
    >>> # contour and below the surface; set the minimum contour z-value to
    >>> # 200, the maximum to 400, and the interval to 50.
    >>> output_dataframe = pygmt.grdvolume(
    ...     grid=grid, contour=[200, 400, 50], output_type="pandas"
```

File: pygmt/src/grdvolume.py [hunk 8]
```diff
@@ -74,7 +76,23 @@ def grdvolume(grid, output_type="pandas", outfile=None, **kwargs):
...
+    ... )
     """
```
Comment (weiji14) [inline pygmt/src/grdvolume.py:96]: Ideally, we would have nice column names like in `grdhisteq`:

https://github.com/GenericMappingTools/pygmt/blob/d85bfdfc212ae5acf36921acdfd92c139288f30a/pygmt/src/grdhisteq.py#L301-L309

But for the sake of time, and just finish this inline example, let's go with this:

```suggestion
    >>> print(output_dataframe)
             0             1             2           3
        0  200  2.144285e+12  7.972228e+14  371.789489
        1  250  2.104042e+12  6.908183e+14  328.329232
        2  300  2.014978e+12  5.877195e+14  291.675420
        3  350  1.892109e+12  4.897545e+14  258.840510
        4  400  1.744792e+12  3.988316e+14  228.584026
        """
```

**Pull request merged**